cpp/src/arrow/util/tracing_internal.h

```diff
@@ -146,6 +158,17 @@ opentelemetry::trace::StartSpanOptions SpanOptionsWithParent(
         return st;                                                                \
       })
 
+#define PROPAGATE_SPAN_TO_GENERATOR(generator)                                \
```

Comment:
I accept your rationale for these macros but could we add a brief comment with that rationale above these one-liner macros?  Otherwise I'm afraid I'll end up asking you yet again :laughing:

Reply:
> Propagate -> The generator "is a part of a larger activity"
Yes, in this case, the span is not ended if the generator finishes. It is just needed so that any spans created in the asynchronously running code (possibly in other threads) do not end up becoming orphan spans in a separate trace.

Reply:
I added a comment here 567fa3004dfad105c6d9af1929bf231052f167f9